File: downloader/status_manager.py
from typing import Dict


class StatusManager:
    def __init__(self):
        self.status: Dict[str, str] = {}

    def update_status(self, url: str, status: str) -> None:
        self.status[url] = status

    def get_status(self, url: str) -> str:
        return self.status.get(url, 'Pending')

    def remove_status(self, url: str) -> None:
        if url in self.status:
            del self.status[url]

File: src/main/downloader/managers/status_manager.ts
import { EventEmitter } from 'events'

class StatusManager extends EventEmitter {
  private status: { [url: string]: string }

  constructor() {
    super()
    this.status = {}
  }

  updateStatus(url: string, status: string): void {
    this.status[url] = status
    this.emit('statusChange', { url, status })
  }

  getStatus(url: string): string {
    return this.status[url] || 'Pending'
  }

  removeStatus(url: string): void {
    delete this.status[url]
    this.emit('statusChange', { url, status: 'Removed' })
  }
}

export { StatusManager }
